### ISSUE TITLE
Select Refactor

### ISSUE THREAD

**Issue by jamesmfriedman** [jamesmfriedman/rmwc#106]
Currently the Select component uses the MDCSelectFoundation::namedItem to get the selected item and value. This has the following limitations:
- You can only have string values
- there is a bug that happens when your value is null
- The select options are formatted in render which *could* potentially become a performance issue if the options array becomes too large and renders too often. This is just a perceived future issue, not a documented one.
- Since values can only be string based, and MDC does DOM lookups for the item, all values are coerced to strings which will make `1 == "1"` truthy.
- MDCSelectFoundation::namedItem is traversing the entire array of options which is another potential choke point.

The goal is to refactor this to be value driven. This has the following implications:
- formatting select options can be done once when the options changed and cached for better performance
- the withMDC HOC needs to be refactored to allow for arbitrary props to be added so that the formatting of options can be moved into the HOC logic. This might be as easy as taking the return of componentWillUpdate and treating it as spreadProps for the wrapped component.
- the select can then be totally value driven and MDCSelectFoundation::selectedIndex can be used.

**In Summary**
  - This gives us value driven selects with strict equality
  - This avoids excessive calls in RMWC to build the options list on updates
  - This avoids a secondary loop in MDC over the select items when setting the value

**Comment by jamesmfriedman** [comment]
As of 1.5, this is no longer relevant since Select is directly integrated with the foundation and MDC is now using native select elements.
